docs/advanced/working_with_git.rst

```diff
@@ -0,0 +1,207 @@
++-------------------------------+--------------------------------------------------------------------------------------+
+| ``git revert <commit-SHA-1>`` |`Revert <http://git-scm.com/docs/git-revert>`_ changes in a commit                    |
++-------------------------------+--------------------------------------------------------------------------------------+
+|``git fetch <remote> <branch>``| The git `fetch <https://www.atlassian.com/git/tutorials/syncing/git-fetch/>`_ command|
+|                               | imports commits or `tags <http://git-scm.com/book/en/v2/Git-Basics-Tagging>`_        |
+|                               | from a remote repository into your local repo.                                       |
++-------------------------------+--------------------------------------------------------------------------------------+
+| **git pull**                  | **Updates** your repo. Shorthand for ``git fetch`` followed by                       |
+|                               | ``git merge FETCH_HEAD``. Recommended                                                |
+|                               | to use it with `- -rebase <http://bit.ly/1yAi5bf>`_.                                 |
++-------------------------------+--------------------------------------------------------------------------------------+
+| **git log**                   | Lists commits made in the current **branch** of the                                  |
+|                               | repo. **Hacks**: ``git log --pretty=oneline``.                                       |
+|                               | Also check `this <https://coderwall.com/p/euwpig/a-better-git-log>`_.                |
++-------------------------------+--------------------------------------------------------------------------------------+
+| **git rebase** :              |  - Rebasing is the process of moving a **branch**                                    |
+|  - ``git rebase <base>``      |    to a new **base commit**                                                          |
+|  - ``git rebase -i <base>``   |  - Interactive `rebase <https://help.github.com/articles/using-git-rebase/>`_        |
+|  - ``git rebase -i HEAD~NUM`` |  - Modifying to a `head <https://help.github.com/articles/about-git-rebase/>`_       |
+|  - ``git rebase -i bbc643cd^``|  - Modify to specified commit bbc643cd                                               |
+|  - ``git rebase --abort``     |  - Abort a rebase                                                                    |
+|  - ``git reflog``             |  - Tracks the changesets to the tip of branch                                        |
++-------------------------------+--------------------------------------------------------------------------------------+
+
+Issues with Pull Requests
+=========================
+
+Problems new contributors face
```

Comment:
I would tend to encourage new contributors here. I recommend changing title from "Problems..." to "Helpful tips for new contributors"